### PR TITLE
#5: Added support for more than one Image/URL in a single bot invocation

### PR DESCRIPTION
**Pull request by lucastarche**
Added several functions (get_urls, get_urls_from_message, get_urls_from_string and get_attachment_urls), in favor of some modularity. Each sauce is sent in a different message.

This fixes issue #5.

### CODE REVIEW

File: src/commands/sauce.rs [hunk 1]
```diff
@@ -87,32 +49,34 @@ pub async fn sauce(ctx: &Context, msg: &Message, args: Args) -> CommandResult {
...
-                    url = sauce.items[i].link
-                ));
-            }
+    for url in &urls {
+        let sauce = saucenao.check_sauce(url.as_str()).await;
+
+        match sauce {
+            Ok(sauce) => {
+                msg.react(ctx, ReactionType::Unicode("✅".into())).await?;
+                info!("URL?: {}", sauce.original_url);
```
Comment (RealKC) [inline src/commands/sauce.rs:62]: Please enclose the URL in <>

File: src/commands/sauce.rs [hunk 2]
```diff
@@ -121,15 +85,106 @@ pub async fn sauce(ctx: &Context, msg: &Message, args: Args) -> CommandResult {
...
+
+    if let Some(mut urls) = msg
+        .referenced_message
+        .clone()
+        .and_then(|m| get_attachment_urls(&*m))
+    {
+        res.append(&mut urls);
+    }
+
+    if res.is_empty() {
```
Comment (RealKC) [inline src/commands/sauce.rs:120]: This won't get the sauce of an attachment if the user has an URL in their message, which is not what we want

**Review by RealKC — CHANGES_REQUESTED** [review]
Please squash your commits :upside_down_face: 

Would be nice if the commit header was more informative too, this one is kinda misleading, making people think the bot can now get sauce from other websites than saucenao

Edit: also pls solve the merge conflicts

File: src/commands/sauce.rs (hunk 1 above)
Comment (RealKC) [reply src/commands/sauce.rs:62]: Was drunk here ig

File: src/commands/sauce.rs (hunk 2 above)
Comment (RealKC) [reply src/commands/sauce.rs:120]: Misunderstood the logic, please add a comment somewhere in this function

Comment (lucastarche) [reply src/commands/sauce.rs:120]: Added relevant comments

**Review by RealKC — APPROVED** [review]

**Pull request merged**